### PR TITLE
Create PEL without any callout’s

### PR DESCRIPTION
**Pull request by branupama**
This commit adds the changes for implementing async PEL creation API. This API doesn’t ask for any callouts information needs to be provided.

### CODE REVIEW

File: src/event_logger.cpp [hunk 1]
```diff
@@ -110,4 +110,76 @@ void EventLogger::createAsyncPelWithI2cBusCallout(
...
+        sd_bus_default(&l_sdBus);
+
+        auto l_rc = -1;
+        if (!i_userData2)
+        {
+            l_rc = sd_bus_call_method_async(
+                l_sdBus, NULL, constants::eventLoggingServiceName,
+                constants::eventLoggingObjectPath,
+                constants::eventLoggingInterface, "Create", NULL, NULL,
+                "ssa{ss}", l_message.c_str(), l_severity.c_str(), 4, "FileName",
```
Comment (souvik1914581) [inline src/event_logger.cpp:155]: Can magic number 4 be described by a `constexpr ` or `macro`?

File: src/event_logger.cpp [hunk 2]
```diff
@@ -110,4 +110,76 @@ void EventLogger::createAsyncPelWithI2cBusCallout(
...
+                "InternalRc", i_internalRc.c_str(), "DESCRIPTION",
+                l_description.c_str());
+        }
+        else
+        {
+            l_rc = sd_bus_call_method_async(
+                l_sdBus, NULL, constants::eventLoggingServiceName,
+                constants::eventLoggingObjectPath,
+                constants::eventLoggingInterface, "Create", NULL, NULL,
+                "ssa{ss}", l_message.c_str(), l_severity.c_str(), 5, "FileName",
```
Comment (souvik1914581) [inline src/event_logger.cpp:166]: Can magic number 5 be described by a `constexpr`  or `macro`?

File: src/event_logger.cpp [hunk 3]
```diff
@@ -110,4 +110,76 @@ void EventLogger::createAsyncPelWithI2cBusCallout(
...
+                            constants::STR_CMP_SUCCESS))
+        {
+            l_description = (*i_userData1).second;
+        }
+
+        sd_bus* l_sdBus = nullptr;
+        sd_bus_default(&l_sdBus);
+
+        auto l_rc = -1;
+        if (!i_userData2)
```
Comment (PriyangaRamasamy) [inline src/event_logger.cpp:149]: what if the API sends both user data 1 and user data 2?

Comment (PriyangaRamasamy) [inline src/event_logger.cpp:149]: why we need both userData 1 and userData2? 

userData1 is an std::pair which can hold n number of pairs. so can we stick with one userData only?

**Review by PriyangaRamasamy — CHANGES_REQUESTED** [review]

File: src/event_logger.cpp (hunk 1 above)
Comment (branupama) [reply src/event_logger.cpp]: keep it as const local variable.

File: src/event_logger.cpp (hunk 2 above)
Comment (branupama) [reply src/event_logger.cpp]: kept it as const local variable.

File: src/event_logger.cpp (hunk 3 above)
Comment (branupama) [reply src/event_logger.cpp]: API params got changed. new patch set has uploaded

Comment (branupama) [reply src/event_logger.cpp]: userData1 and userData2 are kept if anything comes in the future, caller can use these to send more additional data.

File: src/event_logger.cpp [hunk 4]
```diff
@@ -171,4 +171,68 @@ void EventLogger::createAsyncPelWithI2cBusCallout(
...
+        if (l_rc < 0)
+        {
+            logging::logMessage(
+                "Error calling sd_bus_call_method_async, Message = " +
+                std::string(strerror(-l_rc)));
+        }
+    }
+    catch (const sdbusplus::exception::SdBusError& l_ex)
+    {
+        logging::logMessage("createAsyncPel call failed with an error: " +
```
Comment (PriyangaRamasamy) [inline src/event_logger.cpp:234]: User doesn't know what is createAsyncPel. Can you have it like - Asynchronous PEL creation failed : l_ex.what()

**Review by PriyangaRamasamy — CHANGES_REQUESTED** [review]

File: src/event_logger.cpp [hunk 5]
```diff
@@ -171,4 +171,68 @@ void EventLogger::createAsyncPelWithI2cBusCallout(
...
+                                 const std::optional<std::string> i_procedure)
+{
+    (void)i_symFru;
+    (void)i_procedure;
+    try
+    {
+        if (m_errorMsgMap.find(i_errorType) == m_errorMsgMap.end())
+        {
+            throw std::runtime_error(
+                "Error type not found in the error message map to create PEL");
```
[the remark this answers is not in the document]
Comment (branupama) [reply src/event_logger.cpp]: done

File: src/event_logger.cpp [hunk 6]
```diff
@@ -171,4 +171,68 @@ void EventLogger::createAsyncPelWithI2cBusCallout(
...
+            ((!i_description.empty() ? i_description : "VPD generic error"));
+
+        const std::string l_userData1 = ((i_userData1) ? (*i_userData1) : "");
+
+        const std::string l_userData2 = ((i_userData2) ? (*i_userData2) : "");
+
+        sd_bus* l_sdBus = nullptr;
+        sd_bus_default(&l_sdBus);
+
+        const uint8_t l_additionalDataCount = 6;
```
[the remark this answers is not in the document]
Comment (branupama) [reply src/event_logger.cpp:213]: its a count of additional data, we are passing.

File: src/event_logger.cpp (hunk 4 above)
Comment (branupama) [reply src/event_logger.cpp]: updated.

File: include/event_logger.hpp [hunk 7]
```diff
@@ -105,6 +103,34 @@ class EventLogger
...
+     * @param[in] i_description - Error description.
+     * @param[in] i_userData1 - Additional user data [optional].
+     * @param[in] i_userData2 - Additional user data [optional].
+     * @param[in] i_symFru - Symblolic FRU callout data [optional].
+     * @param[in] i_procedure - Procedure callout data [optional].
+     *
+     * @todo: Symbolic FRU and procedure callout needs to be handled in this
+     * API.
+     */
+    static void createAsyncPel(const types::ErrorType& i_errorType,
```
Comment (SunnySrivastava1984) [inline include/event_logger.hpp:123]: Why not name like other APIs
createAsyncPelWithoutInventoryCallout?

File: src/event_logger.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply src/event_logger.cpp:213]: Then please define in constants and give a meaningful name.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/event_logger.hpp (hunk 7 above)
Comment (branupama) [reply include/event_logger.hpp:123]: other API's are representing "WithInventoryCallout" , "WithI2cDeviceCallout" ,  "WithI2cBusCallout"
this API is not taking any of these input for callouts. So kept the simple name.

File: src/event_logger.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (branupama) [reply src/event_logger.cpp:213]: earlier I was passing 6 to sd_bus_call_method_async() directly, bcz of the comment
I created the variable for this. 

But this number can change as it is not constant.
If API changes to handle only for the given "UserData1*" or adds anything new as additional data, this number will get change.
Is it better keep this as local variable in this API ?

[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply src/event_logger.cpp:213]: Use constants::Value_6 and have a comment why 6 which can change later on as well.

File: include/event_logger.hpp (hunk 7 above)
Comment (SunnySrivastava1984) [reply include/event_logger.hpp:123]: Then why the brief gives that information explicitly? 
So if this API is explicitly meant to log PEL without call out, why the name should not represent that?

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/event_logger.hpp (hunk 7 above)
Comment (branupama) [reply include/event_logger.hpp:123]: updated the doxygen.

File: src/event_logger.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (branupama) [reply src/event_logger.cpp]: Done

**Review by SunnySrivastava1984 — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**